### PR TITLE
fix infinite loop when input files is corrupted

### PR DESCRIPTION
**Pull request by yverbin**
This PR fixes infinite  loops while corrupted container logs processing
https://discuss.elastic.co/t/filebeat-loops-while-corrupted-logs-processing/183091

### CODE REVIEW

**Comment by kvch** [comment]
jenkins test this

File: filebeat/input/log/harvester.go [hunk 1]
```diff
@@ -260,6 +260,16 @@ func (h *Harvester) Run() error {
...
 		message, err := h.reader.Next()
+
+		// Get copy of state to work on
+		// This is important in case sending is not successful so on shutdown
+		// the old offset is reported
+		state := h.getState()
+
+		// always save state.Offset
+		state.Offset += int64(message.Bytes)
+		h.state = state
```
Comment (kvch) [inline filebeat/input/log/harvester.go]: If you move this line here, is there a need for updating it again in line 355?

Comment (yverbin) [reply filebeat/input/log/harvester.go]: it looks like  unneeded to update h.state again. I'm going to remove it and update PR after testing.

**Comment by yverbin** [comment]
> As can be seen in the CI, this breaks a large set of tests. My guess is that the offset is not reported correctly anymore.

Oh, I see. But in my case with docker logs this fix works perfectly on some subset of nodes.

> 
> I think we need to take here a different approach then just always updating the state on error. Instead I would rather like to keep the current default behaviour and introduce a config option which could be something like `skip_faulty_line: true` (false by default).
> 
> Then instead of returning here https://github.com/elastic/beats/pull/12443/files#diff-65346f572111bd777e9f934d721a58d7R296 it would just continue (if that works).
> 
> In any case we should have some system tests with a faulty line as part of this PR to confirm that the changes work as expected. Could you also please add a changelog?

I'm going to go back to this issue in about a week and try to find other solutions. Unfortunately this issue is serious. I think I should close this PR, right ?

**Comment by faec** [comment]
It looks like this PR has stalled without a working fix. I've filed an issue for the bug at https://github.com/elastic/beats/issues/16695 so it doesn't get lost. Please let us know if you want to revive this pull request, otherwise it will be autoclosed soon. Thanks!

**Comment by andresrc** [comment]
Closing this due to lack of activity. Please feel free to reopen if you plan to continue working on it. Thanks

**Pull request closed without merging**